webapp/channels/src/components/team_controller/team_controller.tsx

```diff
@@ -213,6 +213,8 @@ function TeamController(props: Props) {
         return null;
     }
 
+    const teamLoaded = team?.name.toLowerCase() === teamNameParam?.toLowerCase();
+
```

Comment:
I lack some expertise in the webapp, so let me ask: is this the usual way of checking that the team has loaded?

Reply:
It is not. The idea here is to handle the following behavior:

- I change the url, so the team I have in the state is no longer the team I should be showing
- Since the url changed, I call `joinTeamOrRedirect` that may take time
- In the meanwhile, I should be in a loading state
- The moment `joinTeamOrRedirect` finish, the state will change, and put back the team into the state.

Reply:
Exactly. You switch to a team you are not a member of, so you need some time to call the proper "join team" function without anything else breaking.